tests/modules/media_avportal_mock/src/AvPortalClientMiddleware.php

```diff
@@ -36,17 +37,27 @@ class AvPortalClientMiddleware {
    */
   protected $eventDispatcher;
 
+  /**
+   * The module extension list.
+   *
+   * @var \Drupal\Core\Extension\ModuleExtensionList
+   */
+  protected ModuleExtensionList $moduleExtensionList;
+
   /**
    * AvPortalClientMiddleware constructor.
    *
    * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
    *   The config factory.
    * @param \Symfony\Component\EventDispatcher\EventDispatcherInterface $eventDispatcher
    *   The event dispatcher.
+   * @param \Drupal\Core\Extension\ModuleExtensionList|null $moduleExtensionList
+   *   The module extension list.
    */
-  public function __construct(ConfigFactoryInterface $configFactory, EventDispatcherInterface $eventDispatcher) {
+  public function __construct(ConfigFactoryInterface $configFactory, EventDispatcherInterface $eventDispatcher, ModuleExtensionList $moduleExtensionList = NULL) {
     $this->configFactory = $configFactory;
     $this->eventDispatcher = $eventDispatcher;
+    $this->moduleExtensionList = $moduleExtensionList ?? \Drupal::service('extension.list.module');
```

Comment:
When is this needed? The service seems to be always available.

Reply:
It's to add a layer of backwards compatibility for users that extended this service.